common/platform.h

```diff
@@ -38,8 +38,15 @@ namespace Common {
 class String;
 
 // Helper macros to get enum value for the platform
+#ifdef _MSC_VER
+// Extra level of indirection required to force macro expansion
```

Comment:
There's no point in keeping two code paths. Keep the MSVC one, since it works in other compilers and add a comment explaining the need for the extra level of indirection